corehq/apps/hqwebapp/forms.py

```diff
@@ -48,22 +51,33 @@ def clean(self):
         if settings.ADD_CAPTCHA_FIELD_TO_FORMS:
             if not self.cleaned_data.get('captcha'):
                 raise ValidationError(_("Please enter valid CAPTCHA"))
-
-        try:
-            cleaned_data = super(EmailAuthenticationForm, self).clean()
-        except ValidationError:
-            user = CouchUser.get_by_username(username)
-            if user and user.is_locked_out():
-                metrics_counter('commcare.auth.lockouts')
-                raise ValidationError(LOCKOUT_MESSAGE)
-            else:
-                raise
         user = CouchUser.get_by_username(username)
         if user and user.is_locked_out():
             metrics_counter('commcare.auth.lockouts')
             raise ValidationError(LOCKOUT_MESSAGE)
+
+        cleaned_data = super(EmailAuthenticationForm, self).clean()
+        if not user:
+            consumer_user = ConsumerUser.objects.get_or_none(user=self.user_cache)
+            if consumer_user:
+                EmailAuthenticationForm.create_web_user_from_consumer_user(username,
+                                                                           password,
+                                                                           get_ip(self.request))
         return cleaned_data
 
+    @staticmethod
+    def create_web_user_from_consumer_user(username, password, ip=None, is_domain_admin=True):
+        now = datetime.utcnow()
+
+        new_user = WebUser.create(None, username, password, None, USER_CHANGE_VIA_WEB, is_admin=is_domain_admin)
```

Comment:
how does this work without a domain? Or I guess having this set as `None` is this reason you get the "please create a domain" after you make this account?

Reply:
![Screenshot from 2021-03-10 19-26-10](https://user-images.githubusercontent.com/47347156/110640316-b4215580-81d6-11eb-8048-5c1a40f6b9ca.png)

Reply:
Once the consumer user with out corresponding couch_user is logged in. He sees the above screen